lib/webpack-cli.js

```diff
@@ -72,89 +82,195 @@ class WebpackCLI extends GroupHelper {
+                    const HelpGroup = require('./groups/HelpGroup');
+                    this.helpGroup = new HelpGroup();
+                    break;
+                }
+                case groups.OUTPUT_GROUP: {
+                    const OutputGroup = require('./groups/OutputGroup');
+                    this.outputGroup = new OutputGroup(value);
+                    this.groups.push(this.outputGroup);
+                    break;
+                }
             }
-            this.groups.push(GroupInstance);
         }
     }
 
-    async runOptionGroups() {
-        const groupResult = {
-            options: {},
-            outputOptions: {},
-            processingMessageBuffer: [],
-        };
-
-        const tmpGroupResult = this.groups.map(Group => Group.run()).filter(e => e);
-        const resolvedResults = await Promise.all(tmpGroupResult);
-        resolvedResults.forEach((e, idx) => {
-            let groupObject = resolvedResults[idx + 1];
-            if (e.processingMessageBuffer) {
-                groupResult.processingMessageBuffer = groupResult.processingMessageBuffer.concat(...e.processingMessageBuffer);
-            }
-            if (!groupObject) {
-                groupObject = {
-                    outputOptions: {},
-                    options: {},
-                };
-            }
-            if (!groupObject.outputOptions) {
-                groupObject.outputOptions = {};
-            }
-
-            groupResult.outputOptions = Object.assign(groupObject.outputOptions, e.outputOptions);
-            if (Array.isArray(e.options)) {
-                const defaultArrayOptions = e.options.map(arrayConfigObject => {
-                    return webpackMerge(groupResult.options, arrayConfigObject);
-                });
-                groupResult.options = defaultArrayOptions;
-                return;
-            }
-            if (Array.isArray(groupResult.options)) {
-                const defaultArrayOptions = groupResult.options.map(arrayConfigObject => {
-                    return webpackMerge(e.options, arrayConfigObject);
+    /**
+     * Responsible to override webpack options.
+     * @param {Object} options The options returned by a group helper
+     * @param {Object} strategy The strategy to pass to webpack-merge. The strategy
+     * is implemented inside the group helper
+     * @private
+     * @returns {void}
+     */
+    _mergeOptionsToConfiguration(options, strategy) {
+        /**
+         * options is an array (multiple configuration) so we create a new
+         * configuration where each element is individually merged
+         */
+        if (Array.isArray(options)) {
+            this.compilerConfiguration = options.map(configuration => {
+                if (strategy) {
+                    return webpackMerge.strategy(strategy)(this.compilerConfiguration, configuration);
+                } else {
+                    return webpackMerge(this.compilerConfiguration, configuration);
```

Comment:
```js
if () {
  return ...
}
return ...
```
The `else` block is redundant. We've been following the above approach across the codebase.